### ISSUE TITLE
[BUG] In system settings defined vlan interface is not configured on nodes

### ISSUE THREAD

**Issue by alexdepalex** [harvester/harvester#1739]
**Describe the bug**
The default network interface of the vlan network is not configured on the nodes. Trying to change the setting after deployment in `Settings` seems to be broken.

**To Reproduce**
1. Add the following to the harvester installer config:
```
system_settings:
  vlan: public
``` 
2. After deployment, the interface name `public` is set in ` Settings`  -> `vlan`, but it's not applied to any nodes.
3. Tring to change the setting in `Settings` through the UI doesn't work.

**Expected behavior**
The default network interface of the vlan network should be configured on all nodes.

**Support bundle**

**Environment:**
 - Harvester ISO version: 
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**
- The UI seems to work correctly, but when editing the setting, it seems broken.
![image](https://user-images.githubusercontent.com/713019/147232097-b0a2b056-0cdf-4c69-941a-d54f611ccdb3.png)
![image](https://user-images.githubusercontent.com/713019/147232151-fccd4850-fc15-4e21-a399-f323b89cb178.png)

- The logs from the network controller also seem to lack the interface name
```
node1:~ # kubectl logs -f pod/harvester-network-controller-lt75f -n harvester-system
I1223 10:41:46.475679       1 main.go:112] Starting network controller with 2 threads.
I1223 10:41:46.475711       1 main.go:117] Starting network controller in namespace: harvester-system.
W1223 10:41:46.571661       1 client_config.go:614] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2021-12-23T10:41:50Z" level=info msg="Do not update the existing CRD NetworkAttachmentDefinition"
time="2021-12-23T10:41:50Z" level=info msg="Do not update the existing CRD NodeNetwork"
I1223 10:41:52.375159       1 monitor.go:52] Start Monitor
time="2021-12-23T10:41:52Z" level=info msg="Starting k8s.cni.cncf.io/v1, Kind=NetworkAttachmentDefinition controller"
time="2021-12-23T10:41:52Z" level=info msg="Starting network.harvesterhci.io/v1beta1, Kind=NodeNetwork controller"
I1223 10:41:53.418311       1 controller.go:95] node network configuration node1-vlan has been changed, spec: {Description: NodeName:node1 Type:vlan NetworkInterface:}
I1223 10:41:53.418471       1 controller.go:172] ignore link not found error, details: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1223 10:41:53.419446       1 controller.go:326] couldn't get VLAN, reason: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1223 10:41:53.456395       1 controller.go:95] node network configuration node1-vlan has been changed, spec: {Description: NodeName:node1 Type:vlan NetworkInterface:}
I1223 10:41:53.456515       1 controller.go:172] ignore link not found error, details: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1223 10:41:53.457706       1 controller.go:326] couldn't get VLAN, reason: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1223 10:41:53.562579       1 controller.go:95] node network configuration node1-vlan has been changed, spec: {Description: NodeName:node1 Type:vlan NetworkInterface:}
I1223 10:41:53.562669       1 controller.go:172] ignore link not found error, details: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1223 10:41:53.563621       1 controller.go:326] couldn't get VLAN, reason: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1223 10:41:53.582583       1 controller.go:95] node network configuration node1-vlan has been changed, spec: {Description: NodeName:node1 Type:vlan NetworkInterface:}
I1223 10:41:53.582646       1 controller.go:172] ignore link not found error, details: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1223 10:41:53.583492       1 controller.go:326] couldn't get VLAN, reason: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1223 10:41:53.604405       1 controller.go:95] node network configuration node1-vlan has been changed, spec: {Description: NodeName:node1 Type:vlan NetworkInterface:}
I1223 10:41:53.604493       1 controller.go:172] ignore link not found error, details: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1223 10:41:53.605441       1 controller.go:326] couldn't get VLAN, reason: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1223 10:41:53.610315       1 controller.go:95] node network configuration node1-vlan has been changed, spec: {Description: NodeName:node1 Type:vlan NetworkInterface:}
I1223 10:41:53.610417       1 controller.go:172] ignore link not found error, details: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1223 10:41:53.611858       1 controller.go:326] couldn't get VLAN, reason: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
```

**Comment by yaocw2020** [comment]
The `ClusterNetwork` and `NodeNetwork` are responsible for the VLAN configurations, not the settings, so the VLAN configuration in the settings would not take effect.

**Comment by TachunLin** [comment]
Verified `not fixed` on `master-bfd6a8a9-head` (02/21). 
Since required [harvester-installer/pull/231](https://github.com/harvester/harvester-installer/pull/231) to be merged. Move to the implement stage 

### Result
Add the following content in to config-create.yaml to install harvester cluster 
```
cluster_networks:
  vlan:
    enable: true
    description: "#1739"
    config:
      defaultPhysicalNIC: eno49
```
After installation, the cluster_network `vlan` not yet set to enabled
```
rancher@hpd8s7:~> sudo -i kubectl get clusternetworks.network.harvesterhci.io vlan -o yaml
apiVersion: network.harvesterhci.io/v1beta1
enable: false
kind: ClusterNetwork
metadata:
  creationTimestamp: "2022-02-22T09:29:48Z"
  finalizers:
  - wrangler.cattle.io/harvester-clusternetwork-controller
  generation: 1
  name: vlan
  resourceVersion: "3288"
  uid: e561049a-c915-47eb-baff-561c28d4cabe
```

### Test Information
* Test Environment: 3 nodes harvester on provo bare machine
* Harvester version: master-bfd6a8a9-head (02/21)

**Comment by TachunLin** [comment]
Verified **fixed** on `master-dba85940-head` (3/3) for both bare metal and kvm. Close this issue.


### Result
* Can correctly create vlan cluster network from harvester create config. 

* Can apply the vlan to all nodes 
![image](https://user-images.githubusercontent.com/29251855/156595844-ab68d44c-d082-424d-9b61-654c44124d86.png)

* Can edit the vlan settings without error
![image](https://user-images.githubusercontent.com/29251855/156592846-5980b074-a748-41e5-89c9-47f8e7a43154.png)


### Test Information
* Test Environment1: 3 nodes harvester on provo bare machine
* Test Environment2: 3 nodes harvester on local kvm machine
* Harvester version: `master-dba85940-head`

### Verify Steps
1. Clone ipxe vagrant example project
2. Edit `config-create.yaml.j2` under /ansible/roles/harvester/templates
3. Add the following content to the end 
  ```
  cluster_networks:
    vlan:
      enable: true
      description: "#1739"
      config:
        defaultPhysicalNIC: ens5
  ```
4. Build up a 3 nodes harvester cluster 
5. Check vlan network in settings page 
6. Edit config in host page
7. Check network settings
8. ssh to harvester master node 
9. Run command `sudo -i kubectl get clusternetworks.network.harvesterhci.io vlan -o yaml`


### Additional Context
Test plan https://github.com/harvester/tests/issues/161 test result


- [x] Checked no error occurs and finish the installation

- [x] Checked ClusterNetwork resource named vlan is created.
- [x] Check Enabled, Disabled button.
- [x] Check it shows correct interface name in Default Network Interface
  ![image](https://user-images.githubusercontent.com/29251855/156592708-11e3679b-8c79-4555-9135-308b4f759572.png)
  ![image](https://user-images.githubusercontent.com/29251855/156592808-68db220d-9569-4695-bad9-9fa59bdb7848.png)
  ![image](https://user-images.githubusercontent.com/29251855/156592846-5980b074-a748-41e5-89c9-47f8e7a43154.png)

- [x] Checked the fields of ClusterNetwork resource named vlan are correct.
- [x] Result should contain name, enable, config, defaultPhysicalNIC with pre-defined value.
  ```
  rancher@harvester-node-0:~> sudo -i kubectl get clusternetworks.network.harvesterhci.io vlan -o yaml
  apiVersion: network.harvesterhci.io/v1beta1
  config:
    defaultPhysicalNIC: ens5
  description: '#1739'
  enable: true
  kind: ClusterNetwork
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"network.harvesterhci.io/v1beta1","config":{"defaultPhysicalNIC":"ens5"},"description":"#1739","enable":true,"kind":"ClusterNetwork","metadata":{"annotations":{},"name":"vlan"}}
    creationTimestamp: "2022-03-03T13:56:44Z"
    finalizers:
    - wrangler.cattle.io/harvester-clusternetwork-controller
    generation: 2
    name: vlan
    resourceVersion: "3189"
    uid: 36ab2525-7d64-44a5-826c-d0ab76ffb33a
  ```
